compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -4207,13 +4211,18 @@ private BLangMatchPattern transformMatchPattern(Node matchPattern) {
                     (BLangMappingMatchPattern) TreeBuilder.createMappingMatchPattern();
             bLangMappingMatchPattern.pos = matchPatternPos;
 
-            for (FieldMatchPatternNode fieldMatchPatternNode : mappingMatchPatternNode.fieldMatchPatterns()) {
+            SeparatedNodeList<FieldMatchPatternNode> fieldMatchPatterns = mappingMatchPatternNode.fieldMatchPatterns();
+            int fieldMatchPatternListSize = fieldMatchPatterns.size();
+            for (int i = 0; i < fieldMatchPatternListSize - 1; i++) {
                 bLangMappingMatchPattern.fieldMatchPatterns.add((BLangFieldMatchPattern)
-                        transformMatchPattern(fieldMatchPatternNode));
+                        transformMatchPattern(fieldMatchPatterns.get(i)));
             }
-            if (mappingMatchPatternNode.restMatchPattern().isPresent()) {
-                bLangMappingMatchPattern.restMatchPattern =
-                        (BLangRestMatchPattern) transformMatchPattern(mappingMatchPatternNode.restMatchPattern().get());
+
+            BLangMatchPattern lastMember = transformMatchPattern(fieldMatchPatterns.get(fieldMatchPatternListSize - 1));
+            if (lastMember.getKind() == NodeKind.REST_MATCH_PATTERN) {
+                bLangMappingMatchPattern.restMatchPattern = (BLangRestMatchPattern) lastMember;
+            } else {
+                bLangMappingMatchPattern.fieldMatchPatterns.add((BLangFieldMatchPattern) lastMember);
```

Comment:
We have a duplicate code of `L4188 - L4192`. WDT of extracting this out to a method (if possible)?

Reply:
When extracting out it introduces more checks to the logic. IMO its better to keep it as it is.

Reply:
I mean introducing a new method and have this logic inside it. Then, you can call that method in both place.

Reply:
Yes, when extracting out to a separate method it causes more type checks and casting since they are two different types.

Reply:
yeah. Seems extracting would require type checking and casting.